src/main/java/apoc/agg/Statistics.java

```diff
@@ -42,21 +43,27 @@ public void aggregate(@Name("value") Number value, @Name(value = "percentiles",
                 } else {
                     values.recordValue(value.longValue());
                 }
+                if (minValue == null || minValue.doubleValue() > value.doubleValue()) {
+                    minValue = value;
+                }
+                if (maxValue == null || maxValue.doubleValue() < value.doubleValue()) {
+                    maxValue = value;
+                }
             }
             this.percentiles = percentiles;
         }
 
         @UserAggregationResult
-        public Map<String,Number> result() {
+        public Map<String, Number> result() {
             long totalCount = values != null ? values.getTotalCount() : doubles.getTotalCount();
             boolean empty = totalCount == 0;
-            Map<String,Number> result = new LinkedHashMap<>(percentiles.size()+6);
-            result.put("min",values != null ? (Number)values.getMinValue() : (Number)doubles.getMinValue());
-            result.put("minNonZero",values != null ? (Number)values.getMinNonZeroValue() : (Number)doubles.getMinNonZeroValue());
-            result.put("max",values != null ? (Number)values.getMaxValue() : (Number)doubles.getMaxValue());
-            result.put("total",totalCount);
-            result.put("mean",values != null ? values.getMean() : doubles.getMean());
-            result.put("stdev",values != null ? values.getStdDeviation() : doubles.getStdDeviation());
+            Map<String, Number> result = new LinkedHashMap<>(percentiles.size() + 6);
+            result.put("min", minValue != null ? minValue : 0L);
```

Comment:
I think we can and should put null there if we had no rows.

Reply:
I implemented it this way because before the change there was always a non-null value in the map as the getMinValue()/getMaxValue() from Histogram answer with `long` values (and I always worry about storing null values anyway).
but I don't mind changing this!

Reply:
yes please change.

Reply:
Hi, I did it in the second commit